### PR TITLE
Save sync tips to DB

### PR DESCRIPTION
**Pull request by terencechain**
This PR implements when performing a state transition (ie process block), the latest sync tips can get saved in the DB

### CODE REVIEW

**Review by rauljordan — APPROVED** [review]
Very straightforward, thanks!

**Comment by terencechain** [comment]
Although there's no risk, but let's hold this up until we merge @potuz PRs first

File: beacon-chain/blockchain/process_block.go [hunk 1]
```diff
@@ -501,3 +509,12 @@ func (s *Service) pruneCanonicalAttsFromPool(ctx context.Context, r [32]byte, b
 	}
 	return nil
 }
+
+// Saves synced and validated tips to DB.
+func (s *Service) saveSyncedTipsDB(ctx context.Context) error {
+	tips := s.cfg.ForkChoiceStore.SyncedTips()
+	if len(tips) == 0 {
+		return nil
```
Comment (potuz) [inline beacon-chain/blockchain/process_block.go:517]: Should we return an error here? tips should never be len 0

File: beacon-chain/forkchoice/protoarray/store.go [hunk 2]
```diff
@@ -40,6 +40,13 @@ func New(justifiedEpoch, finalizedEpoch types.Epoch, finalizedRoot [32]byte) *Fo
 	return &ForkChoice{store: s, balances: b, votes: v, syncedTips: st}
 }
 
+// SyncedTips returns the synced and validated tips from the fork choice store.
+func (f *ForkChoice) SyncedTips() map[[32]byte]types.Slot {
+	f.syncedTips.RLock()
+	defer f.syncedTips.RUnlock()
+	return f.syncedTips.validatedTips
```
Comment (potuz) [inline beacon-chain/forkchoice/protoarray/store.go:47]: Doesn't this return the actual map therefore invalidating the Lock?

**Review by potuz — DISMISSED** [review]

**Review by potuz — APPROVED** [review]

**Pull request merged**